### PR TITLE
Some new feature Backgroundless Pins, Journal Image by default, Journal Image Thumbnail and compatibility with FVTT 9

### PR DESCRIPTION
**Pull request by p4535992**
Here I have tried to reduce the changes as much as possible.

I know for a fact that many people use the following three modules always together "pin-cushion", "foundryvtt-journal-thumbnail", "FVTT-Backgroundless-Pins" the last two have not been updated for almost 10 months and it made sense to me (also to avoid a module conflict) update and integrate these features.

I've checked the licenses and it should be fine.

- Add updated version of the module FVTT-Backgroundless-Pins integrated for avoid conflict (and because the module is not touched for 10 months)
- Add the setting to automatically updated by default the note image with the journal image if present
- Add updated version of the module foundryvtt-journal-thumbnail integrated for avoid conflict (and because the module is not touched for 10 months)

The combination of these features is in great demand in investigative games where you want to put objects in the room.

P.S. I add in the PR a prettier configuration based on what I have seen other foundry developers use, consider using it (or if you can please put your own on git) so as to synchronize us when we develop something and prevent git from going crazy with whitespace, tabs, etc.

### CODE REVIEW

**Comment by eclarke12** [comment]
Thanks, I have done a quick review and it looks pretty good! I will pull and make some changes and aim to release an update later this week.

File: .gitignore [hunk 1]
```diff
@@ -116,4 +115,7 @@ dist
 .pnp.*
 
 # Foundry
-foundry.js
\ No newline at end of file
+foundry.js
+/.vscode
+/package-lock.json
```
Comment (Ethaks) [inline .gitignore]: Although this module is not particularly dependent on npm packages, is there a special reason to ignore the lockfile? It's usually recommended to commit it, and npm can use it for version checking even without `npm ci` (and not clutter commits with irrelevant changes).

File: package.json [hunk 2]
```diff
@@ -1,10 +1,12 @@
 {
   "name": "pin-cushion",
-  "version": "1.5.0",
+  "version": "1.5.1",
   "description": "Provides additional functionality around Map Pins",
   "main": "pin-cushion.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "gulp build",
```
Comment (Ethaks) [inline package.json]: Possibly also a question @eclarke12 but does the gulp build process do something actually useful right now? I spy references to combat-carousel in there.

File: package.json [hunk 3]
```diff
@@ -25,6 +27,12 @@
     "gulp": "^4.0.2",
     "gulp-zip": "^5.0.2",
     "jsdoc-to-markdown": "^6.0.1",
-    "node-fetch": "^2.6.1"
+    "node-fetch": "^2.6.1",
```
Comment (Ethaks) [inline package.json]: Just as a quick check: How many of these are used individually? Prettier is clear, but I'm unsure about the rest (and their need to be direct dependencies)

File: pin-cushion.js [hunk 4]
```diff
@@ -338,27 +456,43 @@ class PinCushion {
...
+
     /**
     * Helper function to register settings
     */
     static _registerSettings() {
         game.settings.registerMenu(PinCushion.MODULE_NAME, "aboutApp", {
-            name: "SETTINGS.AboutAppN",
-            label: "SETTINGS.AboutAppN",
-            hint: "SETTINGS.AboutAppH",
+            name: game.i18n.localize("PinCushion.SETTINGS.AboutAppN"),
```
Comment (Ethaks) [inline pin-cushion.js]: Is localizing these strings necessary? I vaguely remember Foundry doing that already on its own in the settings templates.

File: pin-cushion.js [hunk 5]
```diff
@@ -504,6 +675,27 @@ class PinCushionHUD extends BasePlaceableHUD {
     }
 }
 
+class BackgroundlessControlIcon extends ControlIcon {
```
Comment (Ethaks) [inline pin-cushion.js]: I guess one could argue about whether extending the class or wrapping the single changed method is the better thing to do here. I'm not sure about what other modules live in this space though, so I cannot really comment on the necessity to play nice with them.

File: .gitignore (hunk 1 above)
Comment (p4535992) [reply .gitignore]: didn't know that, no problem to add the package-lock.json

File: pin-cushion.js (hunk 4 above)
Comment (p4535992) [reply pin-cushion.js]: not necessary , foundry should be try to localize the label anyway

File: package.json (hunk 2 above)
Comment (p4535992) [reply package.json]: i made a little fix for that

**Comment by eclarke12** [comment]
@p4535992 I haven't forgotten about this. I'm just finishing up a big CUB update then I will look at merging this PR.

**Comment by p4535992** [comment]
No problem man  CUB is much more important. 
By the way I committed a beta without thinking about it, tonight I'll fix it.

**Comment by p4535992** [comment]
K the last commit is a starter integration of the "Point of Interest Teleport" module, for which I wanted to create a series of new features to integrate with pin cushion, so you might as well ignore it if you want. For now I will not update this PR again. Let me know if I need to fix anything for you. Greetings

**Comment by Shuggaloaf** [comment]
Hi all! I actually came to death-save's git to request that he take up development of backgroundless pins because of it being abandoned. (And the pin cushion dev seemed like the logical place to start)

I saw this PR and couldn't believe it! Excellent job @p4535992 thank you so much for not just having the idea but helping make it happen. And you are right, I do also use journal thumbnails as well. Having all 3 in 1 mod, with reduced conflicts and mods count will be very nice. 

Thanks @eclarke12 for grabbing this and incorporating it. I just grabbed the updated CUB and so far it is great! (Thanks for fixing that double concentration issue). I'll be monitoring this for the merge. Looking forward to it.

**Comment by Shuggaloaf** [comment]
@p4535992 / @eclarke12 May possibly want to consider not including the POI Teleport at this time. I say this hesitantly as I am 100% for combining as many mods as possible to reduce possible conflicts.

There are several other existing methods/mods to accomplish this. 
Trigger Happy mod can do this very easily just by adding a transparent drawing over the map pin area. When clicked (or optionally when a token is moved to that area), it loads whatever scene you would like it to. It also is not Scene Notes dependent.
Also, quite simply with no mods, you can just drag a scene into the journal entry of that map pin and create a link. Click to open map pin journal entry, click scene link, done. Usage is pretty much the same amount of clicks with no mod dependencies and less setup. 

Just some thoughts!

**Comment by Ethaks** [comment]
@p4535992, would you consider the PR stable as is, or do you have plans for further additions etc.?
If you consider it to be somewhat done, I'll take another look and possibly change up some stuff/build upon it, before hopefully getting a release ready soon-ish.

**Comment by p4535992** [comment]
@Ethaks my PR was born to integrate some deprecated modules that I always used together (even if now they have updated them for 9) and synchronize everything with the wrapper module. Except for issues that are pointed out to me, I don't think I'll get back on it, for me you can try to take a look at it.

**Comment by p4535992** [comment]
@eclarke12 @Ethaks just a reminder for this, i made some test and seem working fine

**Comment by p4535992** [comment]
Now i'm officially the new maintainer of the module so i'll close this PR

**Comment by p4535992** [comment]
I'll close this pr because, i'm now the officialy new maintainer.

**Pull request closed without merging**